code-challenges.test.js

```diff
@@ -21,9 +21,46 @@ const people = [
 ]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe("objSentence", () => {
+  it("returns an array with a sentence about each person with their name capitalized", () => {
+    expect(objSentence(people)).toEqual(["Ford Perfect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]);
+  })
+})
+
+//  ReferenceError: objSentence is not defined
+//  Note: my code is correct but for some reason the test does not pass. 
+
 
 // b) Create the function that makes the test pass.
 
+// Pseudo Code: 
+// Declare a function called objSentence
+// Name param: Object
+// Return object.map to iterate over the object thats passed in as an argument
+// Use string interpolation to return key value pairs in a sentence.
+// Chain methods .charAt and .toUpperCase to value.name to capitalize the first index of the values of name:
+// Use concatenation (+) to join the previous step to value.name.slice(1).
+
+//-------This is my first attempt 
+// I tried creating the function that makes the test pass before even creating the test which I realized was not a good idea. When I did write the test it did not pass and I was able to quickly realize the disparity between the expected output and the test I had written. This was definitely a great learning experience as it is so easy to overlook things and the order in which we do things matter. 
+// const objSentence = (object) => {
+//     return object.map(value => {
+//       return `${value.name.charAt(0).toUpperCase() + value.name.slice(1)} is ${value.occupation}`
+//       })  
+// }
+
+//--------Passing code
+// Same idea for the pseudo code as my first attempt but with more variables to target the first and last name of the person name:
+
+const objSentence = (object) => {
+  return object.map(value => {
+   let name = value.name.split(" ")
+   let firstName = name[0].charAt(0).toUpperCase() + name[0].slice(1)
+   let lastName = name[1].charAt(0).toUpperCase() + name[1].slice(1)
+   return `${firstName} ${lastName} is ${value.occupation}`
+  })
+}
+// console.log(objSentence(people))
```

Comment:
Awesome job, William!  Your test didn't pass because of 2 minor issues.  #1 - in your expect statement, you had a typo with Ford Prefect's name - You entered Ford Perfect.  #2 - in your return statement in your function, you forgot to end the sentence with a period. Otherwise, your code is solid!  Great work, and thank you for including your attempts and process.